x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -628,6 +628,35 @@ public void testWeightedAvg() {
         );
     }
 
+    public void testMatchInsideEval() throws Exception {
+        assertEquals(
+            "1:36: EVAL does not support MATCH expressions",
+            error("row title = \"brown fox\" | eval x = title match \"fox\" ")
+        );
+    }
+
+    public void testMatchFilter() throws Exception {
+        assertEquals(
+            "1:63: MATCH requires a mapped index field, found [name]",
+            error("from test | eval name = concat(first_name, last_name) | where name match \"Anna\"")
```

Comment:
How about `from test | rename first_name as fn | where fn match \"Anna\"`?

Reply:
Good catch!
Added this as a test - it now fails with: `1:45: MATCH requires a mapped index field, found [fn]`
Definitely something we want to revisit, because we can argue that even if the column gets renamed we should still be able to push this down to the lucene query.
Tracking this in the meta issue: https://github.com/elastic/elasticsearch/issues/110131

Reply:
I'm ok with this being addressed as a follow up.